mime-types/Network/Mime.hs

```diff
@@ -75,7 +81,17 @@ defaultMimeType = "application/octet-stream"
 --
 -- Generated from the Apache and nginx mime.types files.
 defaultMimeMap :: MimeMap
-defaultMimeMap = Map.fromAscList [
+defaultMimeMap = Map.fromAscList mimeAscList
+
+-- | A default mapping from mime type to filename extensions.
+-- Note there can be multiple extensions for a given mime type.
+-- Generated from the Apache and nginx mime.types files.
+-- @since 0.1.2.0
+defaultExtensionMap :: ExtensionMap
+defaultExtensionMap = Map.fromAscList $ map (\xs -> (fst $ head xs, map snd xs)) $ groupBy ((==) `on` fst) . sort . map swap $ mimeAscList
```

Comment:
Would it maybe be an idea to turn this into a `foldl'` and make the `ExtensionMap` have a `Set Extension` as values?

```hs
type ExtensionMap = Map.Map MimeType (Set Extension)

defaultExtensionMap :: ExtensionMap
defaultExtensionMap =
    Data.List.foldl' (flip go) mempty mimeAscList
  where
    go (ext, mime) =
      Map.alter (Just . maybe (Set.singleton ext) (Set.insert ext)) mime
```

Otherwise, with keeping it a list:
```hs
defaultExtensionMap :: ExtensionMap
defaultExtensionMap =
    Data.List.foldl' (flip go) mempty mimeAscList
  where
    go (ext, mime) =
      Map.alter (Just . maybe [ext] (ext :)) mime
```
(or maybe use `foldr` to not have the extensions in reverse order?